src/vmm/src/guest_config/cpuid/mod.rs

```diff
@@ -435,11 +466,26 @@ impl From<Cpuid> for RawCpuid {
     }
 }
 
-impl From<Cpuid> for kvm_bindings::CpuId {
-    #[inline]
-    fn from(cpuid: Cpuid) -> Self {
-        let raw_cpuid = RawCpuid::from(cpuid);
-        Self::from(raw_cpuid)
+impl TryFrom<Cpuid> for kvm_bindings::CpuId {
+    type Error = utils::fam::Error;
+
+    fn try_from(cpuid: Cpuid) -> Result<Self, Self::Error> {
+        let entries = cpuid
+            .inner()
+            .iter()
+            .map(|(key, entry)| kvm_bindings::kvm_cpuid_entry2 {
```

Comment:
A similar point to https://github.com/firecracker-microvm/firecracker/pull/3625/commits/7d472cc9e7af222a9cc92159bd280e0cb0da47c6#r1175133069, it would be better to have a `From` implementation to simplify this code.